src/main/java/org/folio/rest/impl/CodexInstancesImpl.java

```diff
@@ -91,4 +88,22 @@ public void getCodexInstancesById(String id, String lang,
         return null;
       });
   }
+
+  private CompletionStage<InstanceCollection> getCodexInstances(String query, int offset, int limit, Context vertxContext, RMAPIConfiguration rmAPIConfig){
+    try {
+      if (limit == 0 || query == null) {
+        throw new QueryValidationException("Unsupported Query Format : Limit/Query suggests that no results need to be returned.");
+      }
+      final CQLParserForRMAPI parserForRMAPI;
+
+      CQLParameters cqlParameters = new CQLParameters(query);
+      if (cqlParameters.isIdSearch()) {
+        return RMAPIToCodex.getInstanceById(vertxContext, rmAPIConfig, cqlParameters.getIdSearchValue());
+      }
```

Comment:
Just curious why you decided to move this logic (believe it was in RMAPIToCodex before) -- having it here does make it more apparent that a list of 1 is returned

Reply:
I wanted CQLParserForRMAPI to take different types of parameters (TitleParameters, PackageParameters), so I needed to parse cql and do check for id before CQLParserForRMAPI (there are probably better ways to do that), 
This also removes the need to set and check isIdSearch flag in multiple places, previously CQLParserForRMAPI constructor and RMAPIToCodex#getInstances had completely different behaviours based on this flag.
Now the name CQLParserForRMAPI doesn't make much sense, so I will try to refactor it